File: android/src/main/java/com/visioncameramlkit/domain/models/TextRecognitionOptions.kt
package com.visioncameramlkit.domain.models

data class TextRecognitionOptions(
  val language: TextRecognitionLanguage = TextRecognitionLanguage.LATIN,
  val invertColors: Boolean = false,
  // Android handles rotation automatically, so outputOrientation is ignored. Added for iOS parity.
  val outputOrientation: OutputOrientation? = null,
  val scaleFactor: Float = 1.0f,
)

enum class TextRecognitionLanguage {
  LATIN,
  CHINESE,
  DEVANAGARI,
  JAPANESE,
  KOREAN,
}

File: ios/Domain/Models/TextRecognitionOptions.swift
import Foundation

struct TextRecognitionOptions {
  let language: DomainTextRecognitionLanguage
  let invertColors: Bool
  let outputOrientation: OutputOrientation?
  let scaleFactor: CGFloat

  init(
    language: DomainTextRecognitionLanguage = .latin,
    invertColors: Bool = false,
    outputOrientation: OutputOrientation? = nil,
    scaleFactor: CGFloat = 1.0
  ) {
    self.language = language
    self.invertColors = invertColors
    self.outputOrientation = outputOrientation
    self.scaleFactor = scaleFactor
  }
}

enum DomainTextRecognitionLanguage: String {
  case latin = "LATIN"
  case chinese = "CHINESE"
  case devanagari = "DEVANAGARI"
  case japanese = "JAPANESE"
  case korean = "KOREAN"
}
